src/boot/bootctl-status.c

```diff
@@ -76,36 +78,78 @@ static int status_entries(
-                       SD_ID128_FORMAT_VAL(dollar_boot_partition_uuid));
-        if (settle_entry_token() >= 0)
-                printf("\n        token: %s", arg_entry_token);
-        printf("\n\n");
-
-        if (config->default_entry < 0)
-                printf("%zu entries, no entry could be determined as default.\n", config->n_entries);
-        else {
-                printf("%sDefault Boot Loader Entry:%s\n", ansi_underline(), ansi_normal());
-
-                r = show_boot_entry(
-                                boot_config_default_entry(config),
-                                &config->global_addons,
-                                /* show_as_default= */ false,
-                                /* show_as_selected= */ false,
-                                /* show_discovered= */ false);
-                if (r > 0)
-                        /* < 0 is already logged by the function itself, let's just emit an extra warning if
-                           the default entry is broken */
-                        printf("\nWARNING: default boot entry is broken\n");
+        if (arg_json_format_flags != JSON_FORMAT_OFF) {
+                char boot_path[256];
+                if (!sd_id128_is_null(dollar_boot_partition_uuid)) {
+                        snprintf(boot_path, sizeof(boot_path), "%s (/dev/disk/by-partuuid/" SD_ID128_FORMAT_STR ")",
+                                 dollar_boot_path, SD_ID128_FORMAT_VAL(dollar_boot_partition_uuid));
+                }
+                if (settle_entry_token() >= 0)
+                        r = json_variant_merge_objectb(v, JSON_BUILD_OBJECT(
+                                            JSON_BUILD_PAIR("BootLoaderEntries", JSON_BUILD_OBJECT(
+                                                JSON_BUILD_PAIR("$BOOT", JSON_BUILD_STRING(boot_path)),
+                                                JSON_BUILD_PAIR("token", JSON_BUILD_STRING(arg_entry_token))
+                                            ))
+                                        ));
+                else
+                        r = json_variant_merge_objectb(v, JSON_BUILD_OBJECT(
+                                            JSON_BUILD_PAIR("BootLoaderEntries", JSON_BUILD_OBJECT(
+                                                JSON_BUILD_PAIR("$BOOT", JSON_BUILD_STRING(boot_path))
+                                            ))
+                                        ));
+                if (r < 0)
+                        return log_oom();
+
+  
+                _cleanup_(json_variant_unrefp) JsonVariant *b = NULL;
+
+                r = boot_entry_to_json(config, 0, &b);
+                if (r < 0)
+                        return log_oom();
+
+                r = json_variant_merge_objectb(v, JSON_BUILD_OBJECT(
+                                 JSON_BUILD_PAIR_CONDITION(b, "DefaultBootLoaderEntry", JSON_BUILD_VARIANT(b))));
+                if (r < 0)
+                        return log_oom();
+
+        } else {
```

Comment:
so we have this rule that return params should be initialized in all success paths. in the non-json path this means you should initialize `*ret_variant` too. Since in that case you have no JSON object, set it to NULL.